get_mock.go

```diff
@@ -43,3 +43,10 @@ func (g *MockGetter) GetFile(dst string, u *url.URL) error {
 	}
 	return g.GetFileErr
 }
+
+func (g *MockGetter) ClientMode(u *url.URL) (ClientMode, error) {
+	if u.Path[len(u.Path)-1:] == "/" {
```

Comment:
Maybe don't need comprehensive error checking in a mock, but it looks like this will panic if the URL is `""`.